ugvc/pipelines/vcfeval_flavors.py

```diff
@@ -0,0 +1,113 @@
+#!/env/python
+
+# Copyright 2022 Ultima Genomics Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+# DESCRIPTION
+#    Compare vcf file to ground-truth in alternative manner
+# CHANGELOG in reverse chronological order
+
+from __future__ import annotations
+import argparse
+import subprocess
+import json
+import pysam
+
+from simppl.simple_pipeline import SimplePipeline
+from ugvc.utils.stats_utils import get_precision, get_recall, get_f1
+
+
+def get_parser(argv: list[str]):
+    ap = argparse.ArgumentParser(prog='vcfeval_flavors.py', description=run.__doc__)
+    ap.add_argument('-b', '--baseline',
+                        help="VCF file containing baseline variants", type=str, required=True)
+    ap.add_argument('-c', '--calls',
+                        help="VCF file containing called variants--output_prefix", type=str, required=True)
+    ap.add_argument('-e', '--evaluation_regions',
+                        help='if set, evaluate within regions contained in the supplied BED file, '
+                             'allowing transborder matches. To be used for truth-set', type=str)
+    ap.add_argument('-o', '--output',
+                        help='directory for output', type=str, required=True)
+    ap.add_argument('-t', '--template',
+                        help='SDF of the reference genome the variants are called against')
+    ap.add_argument('-p', '--allele_and_genotype_error_penalty',
+                        type=int,
+                        choices=[2, 1, 0, -1],
+                        default=0,
+                        help="-p 2: usual vcfeval, penalizes twice each wrong-allele/genotype errror\n" \
+                             "-p 1: penalize wrong-allele/genotype only once (remove 0.5 of such fps and fns)\n" \
+                             "-p 0: don't penalize wrong-allele/genotype (remove completely such fps and fns)\n" \
+                             "-p -1: don't penalize, and also reward a tp for calling wrong-allele/genotype\n")
+    return ap
+    return args
+
+def count_vcf_lines(vcf_file):
```

Comment:
Consider using `bcftools index -n`, instead (pysam is sloooooooow)

Reply:
great, I was looking for something like that!